### PR TITLE
Add ob-ein tests

### PR DESCRIPTION
**Pull request by dickmao**
Also, speed up login and other cleanups

### CODE REVIEW

File: lisp/ob-ein.el [hunk 1]
```diff
@@ -275,66 +198,87 @@ jupyter kernels.
...
-    (cond ((null path)
-           (let* ((name ein:org-babel-default-session-name)
-                  (new-session (format "%s/%s" url-or-port name)))
-             (ein:notebooklist-new-notebook-with-name name kernelspec url-or-port)
-             (org-babel-ein-initiate-session new-session kernelspec)))
-          (t (let ((nb (ein:org-find-or-open-session session kernelspec)))
-               (ein:$notebook-kernel nb))))))
+(defun org-babel-edit-prep:ein (babel-info)
+  (let ((buffer (current-buffer)))
+    (ob-ein--initiate-session
```
Comment (millejoh) [inline lisp/ob-ein.el]: The signature on this has changed to 3 arguments: session, kernelspec, callback.

**Comment by millejoh** [comment]
Other than that one comment, very slick. Not sure if it is my imagination, but performance overall seems much improved as well.

**Comment by millejoh** [comment]
Okay; I'm happy. You happy? Time to merge?

**Comment by dickmao** [comment]
Not yet.

**Comment by millejoh** [comment]
No problem. I'll wait.

File: lisp/ob-ein.el (hunk 1 above)
Comment (dickmao) [reply lisp/ob-ein.el]: Thanks, that was a real problem.  I added a test for `C-c '` to exercise edit-prep.

**Comment by dickmao** [comment]
The travis situation is not great (in particular, jupyter often gets into a 403 rut both on travis and on my laptop), but I'm removing the WIP status.

**Pull request merged**